code/game/gamemodes/wizard/soulstone.dm

```diff
@@ -106,69 +107,239 @@
+					return
+				else
+					to_chat(new_target, "<span class='danger'>You feel your soul getting sucked into the soulstone.</span>")
+					to_chat(user, "<span class='rose'>The soul stone reacts to the corpse and starts glowing.</span>")
+					capture_soul_process(user,new_target.client,humanHead,humanHead.origin_body)
+		else
+			to_chat(humanBrainMob, "<span class='danger'>You feel your soul getting sucked into the soul stone.</span>")
+			to_chat(user, "<span class='rose'>The soul stone reacts to the corpse and starts glowing.</span>")
+			capture_soul_process(user,humanBrainMob.client,humanHead,humanHead.origin_body)
+
+
+/obj/item/proc/capture_soul_process(var/mob/living/carbon/user, var/client/targetClient, var/atom/movable/target, var/atom/movable/add_target = null)
+	//user is the guy using the soulstone
+	//C is the client of the guy we're soulstoning, so we don't lose track of him between the beginning and the end of the soulstoning.
+	//target is the source of the guy's soul (his body, or his head if decapitated)
+	//add_target is his body if he has been decapitated, for cosmetic purposes (and so it dusts)
+
+	if(!targetClient)
+		return
+
+	var/mob/living/carbon/human/body = null
+
+	if(istype(target,/mob/living/carbon/human))
+		body = target
+	else if(istype(add_target,/mob/living/carbon/human))
+		body = add_target
+
+	var/true_name = "Unknown"
+
+	if(body)
+		true_name = body.real_name
+
+		for(var/obj/item/W in body)
+			body.drop_from_inventory(W)
+
+		body.dropBorers(1)
+
+		var/turf/T = get_turf(body)
+
+		body.invisibility = 101
+
+		var/datum/organ/external/head_organ = body.get_organ("head")
+		if(head_organ.status & ORGAN_DESTROYED)
+			new /obj/effect/decal/remains/human/noskull(T)
+			anim(target = T, a_icon = 'icons/mob/mob.dmi', flick_anim = "dust-h2-nohead", sleeptime = 26)
+		else
+			new /obj/effect/decal/remains/human(T)
+			if(body.lying)
+				anim(target = T, a_icon = 'icons/mob/mob.dmi', flick_anim = "dust-h2", sleeptime = 26)
+			else
+				anim(target = T, a_icon = 'icons/mob/mob.dmi', flick_anim = "dust-h", sleeptime = 26)
+
+		if(body.decapitated && (body.decapitated == target))//just making sure we're dealing with the right head
+			target.invisibility = 101
+			new /obj/item/weapon/skull(get_turf(target))
+	else
+		target.invisibility = 101
+
+		if(ismob(target))
+			var/mob/M = target
+			true_name = M.real_name
+			new /obj/effect/decal/cleanable/ash(get_turf(target))
+		else if(istype(target,/obj/item/weapon/organ/head))
+			var/obj/item/weapon/organ/head/H = target
+			var/mob/living/carbon/brain/BM = H.brainmob
+			true_name = BM.real_name
+			new /obj/item/weapon/skull(get_turf(target))
+
+	//Scary sound
+	playsound(get_turf(src), get_sfx("soulstone"), 50,1)
+
+	//Creating a shade inside the stone and putting the victim in control
+	var/mob/living/simple_animal/shade/shadeMob = new(src)//put shade in stone
+	shadeMob.status_flags |= GODMODE //So they won't die inside the stone somehow
```

Comment:
Make a proc that resets these, and use it to remove a shade from the container of a soul stone or else you get problems like this 
https://github.com/DeityLink/vgstation13/blob/soulblade/code/game/gamemodes/wizard/soulstone.dm#L15